recipes/app3.rb

```diff
@@ -94,3 +95,31 @@
   template 'app-nginx.erb'
   cookbook 'osl-app'
 end
+
+# Docker containers
+mulgara_redmine_creds = data_bag_item('mulgara_redmine', 'mysql_creds')
+
+docker_image 'library/redmine' do
+  tag '4.0.4'
+  action :pull
+end
+
+# Check if attribute is set for testing
+mulgara_db_host = if node['osl-app'].attribute?('mulgara_redmine_mysql_hostname')
+                    node['osl-app']['mulgara_redmine_mysql_hostname']
+                  else
+                    mulgara_redmine_creds['db_hostname']
+                  end
+
+docker_container 'code.mulgara.org' do
+  repo 'redmine'
+  tag '4.0.4'
+  port '8084:3000'
+  restart_policy 'always'
+  env [
+    "REDMINE_DB_MYSQL=#{mulgara_db_host}",
+    "REDMINE_DB_DATABASE=#{mulgara_redmine_creds['db_db']}",
+    "REDMINE_DB_USERNAME=#{mulgara_redmine_creds['db_user']}",
+    "REDMINE_DB_PASSWORD=#{mulgara_redmine_creds['db_passwd']}",
+  ]
+end
```

Comment:
I think we need to consider how we manage the stateful data for the container. If you look at the [container README](https://hub.docker.com/_/redmine), I think I'd like to go with mounting the files folder from somewhere such as ``/data/code.mulgara.org/redmine``. That would also make backups much easier.

In addition, we might want to consider enabling ``REDMINE_NO_DB_MIGRATE`` and/or ``REDMINE_PLUGINS_MIGRATE``.

Reply:
We actually need to migrate the DB and plugins at first, since they are for an older version of redmine. After the first run we can disable it

Reply:
As noted in our conversation earlier this week, I think it'll be good to add this here so that future upgrades go smoothly.